src/factories.js

```diff
@@ -30,23 +34,23 @@ const mergePropsAndClassName = (props, extraProps) => {
  * @param {function|string} Component A ReactClass or string
  * @param {function} mapValueToProps A function that maps a primitive value to the Component props
  * @param {string|object|function} val The value to create a ReactElement from
- * @param {object} extraProps Additional props to add to the final ReactElement
+ * @param {object} defaultProps Default props to add to the final ReactElement
  * @returns {function|null}
  */
-export function createShorthand(Component, mapValueToProps, val, extraProps = {}) {
+export function createShorthand(Component, mapValueToProps, val, defaultProps = {}) {
   // Clone ReactElements
   if (isValidElement(val)) {
-    return React.cloneElement(val, mergePropsAndClassName(val.props, extraProps))
+    return React.cloneElement(val, mergePropsAndClassName(defaultProps, val.props))
   }
 
   // Create ReactElements from props objects
   if (_.isPlainObject(val)) {
-    return <Component {...mergePropsAndClassName(val, extraProps)} />
+    return <Component {...mergePropsAndClassName(defaultProps, val)} />
   }
 
   // Map values to props and create a ReactElement
   if (_.isString(val) || _.isNumber(val)) {
-    return <Component {...mergePropsAndClassName(mapValueToProps(val), extraProps)} />
+    return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
```

Comment:
Wrote out a detailed explanation of how I didn't think this actually work for the string/number literal case only to come to see it correctly.  Thanks for the clarifications!